src/Bicep.Core/TypeSystem/FunctionPlacementFlags.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+
+namespace Bicep.Core.TypeSystem
+{
+    /// <summary>
+    /// Flags that may be placed on functions to indicate where they can be placed in semantic tree
+    /// </summary>
+    [Flags]
+    public enum FunctionPlacementFlags
```

Comment:
>FunctionPlacementFlags [](http://example.com/codeflow?start=16&length=22)

`FunctionFlags` already has a bunch of placement-like flags. Do we need the second enum?

Reply:
Function flags are used in type assignment. adding a flag there and assigning it to function overload will trigger the mechanism and prevent using it - type assignment visitor will want function with default flag no with something different.

Reply:
@majastrz - is this explanation sufficient?

Reply:
Both `FunctionFlags` and `FunctionPlacementFlags` are declared and assigned together in every place they appear, so their lifecycles appear to be identical.

If you look at these values of `FunctionFlags`, all of them control placement of the functions:
- `ParamDefaultsOnly`
- `ParameterDecorator`
- `VariableDecorator`
- `ResourceDecorator`
- `ModuleDecorator`
- `OutputDecorator`
- `ResourceOrModuleDecorator`
- `AnyDecorator`

In fact it seem like the majority of the enum values are about placement, so it's kind of weird to have a second set of flags also about placement.

You mentioned that adding the flag to `FunctionFlag` would trigger the type checking mechanism? Can you point me to the code that's doing it? Maybe we can tweak that to avoid it?

Reply:
it's here: https://github.com/Azure/bicep/blob/de8fd8ed4d4aefd6519fe726953f8f255669e1ae/src/Bicep.Core/Semantics/SymbolValidator.cs#L105

In fact I can see now that additional flag will not impact the binding mechanism, but we will end up in having flags checked in 2 places.

Reply:
@majastrz - as discussed, I merged it into one enum (Function Flags).